lib/puppet/type/mongodb_user.rb

```diff
@@ -4,9 +4,32 @@
   ensurable
 
   def initialize(*args)
+
+    # required to create the password hash from plaintext
+    require 'digest/md5'
+
     super
     # Sort roles array before comparison.
     self[:roles] = Array(self[:roles]).sort!
+    if self[:ensure]
```

Comment:
this logic checks that password_plain and password_hash correspond to each other:
- if there's only a plaintext password, create the hash from it
- if both are present, check if they correspond; if not, fail
- if none is present, fail